stripe/src/main/java/com/stripe/android/StripeKtx.kt

```diff
@@ -570,11 +571,122 @@ suspend fun Stripe.confirmPaymentIntent(
+)
+suspend fun Stripe.getPaymentIntentResult(
+    requestCode: Int,
+    data: Intent?,
+): PaymentIntentResult {
+    return runApiRequestCaptureIllegalArgumentException(
+        isForPaymentIntentResult(
+            requestCode,
+            data
+        )
+    ) { paymentController.getPaymentIntentResult(data!!) }
+}
+
+/**
+ * Get the [SetupIntentResult] from [Intent] returned via
+ * Activity#onActivityResult(int, int, Intent)}} for SetupIntentResult confirmation.
+ * (see [confirmSetupIntent])
+ *
+ * @param requestCode [Int] code passed from Activity#onActivityResult
+ * @param data [Intent] intent from Activity#onActivityResult
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class,
+    IllegalArgumentException::class
+)
+suspend fun Stripe.getSetupIntentResult(
+    requestCode: Int,
+    data: Intent?,
+): SetupIntentResult {
+    return runApiRequestCaptureIllegalArgumentException(
+        isForSetupIntentResult(
+            requestCode,
+            data
+        )
+    ) { paymentController.getSetupIntentResult(data!!) }
+}
+
+/**
+ * Get the [Source] from [Intent] returned via
+ * Activity#onActivityResult(int, int, Intent)}} for [Source] authentication.
+ * (see [authenticateSource])
+ *
+ * @param requestCode [Int] code passed from Activity#onActivityResult
+ * @param data [Intent] intent from Activity#onActivityResult
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.getAuthenticateSourceResult(
+    requestCode: Int,
+    data: Intent?,
+): Source {
+    return runApiRequestCaptureIllegalArgumentException(
+        isAuthenticateSourceResult(
+            requestCode,
+            data
+        )
+    ) { paymentController.getSource(data!!) }
+}
+
+/**
+ * Consume the [IllegalArgumentException] caused by empty result from Stripe's internal Json Parser,
+ * throw [InvalidRequestException] for public API.
+ *
+ * @return the result if the API result and JSON parsing are successful; otherwise, throw an exception.
+ */
+internal inline fun <reified APIObject : StripeModel> runApiRequestCaptureIllegalArgumentException(
```

Comment:
Can we simplify this method name? Is it really relevant what the exception type is?

Reply:
Sure, the doc already explains what it does, renamed to `runApiReques`